dbt_subprojects/dex/models/_projects/fluid/ethereum/fluid_ethereum_pools.sql

```diff
@@ -0,0 +1,52 @@
+{{
+    config(
+        schema = 'fluid_ethereum',
+        alias = 'pools',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['dex'],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')],
+        post_hook='{{ expose_spells(blockchains = \'["ethereum"]\',
+                                    spell_type = "project", 
+                                    spell_name = "fluid", 
+                                    contributors = \'["maybeYonas", "pyor_xyz"]\') }}'
+    )
+}}
+
+{% set weth_address = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' %}
+
+with
+decoded_events as (
+    select 
+        block_time,
+        block_number,
+        index as evt_index,
+        tx_hash,
+        contract_address as factory,
+        substr(topic1, 13) as dex,
+        substr(topic2, 13) as supplyToken,
+        substr(topic3, 13) as borrowToken,
+        bytearray_to_uint256(data) as dexId
+    from {{ source('ethereum', 'logs')}}
+    where topic0 = 0x3fecd5f7aca6136a20a999e7d11ff5dcea4bd675cb125f93ccd7d53f98ec57e4 
+    -- DexT1Deployed -> sample tx: https://etherscan.io/tx/0xabf5c0e676e69de941c283400d7ac5f47b17a09d870f225b5240522f95da501c#eventlog
```

Comment:
any particular reason we're not going through normal decoding process here, to build those tables vs. reading raw data?

Reply:
the etherscan abi didn't have that particular event. 
i can manually edit the json to add that single event in. 
I'm not sure if its safe though. 

https://etherscan.io/address/0x91716C4EDA1Fb55e84Bf8b4c7085f84285c19085#code

Reply:
i see a warning there that the contract may be a proxy? doesn't that typically mean you can find the underlying contract, and that has the correct ABI (or am i oversimplifying that)?

Reply:
cant really say its the correct ABI, since the events are split between both proxy and impl.
i think the ideal work around is to manually add the event to the abi json. i however suspect, with their v2 launch this could change. 

https://github.com/Instadapp/fluid-contracts-public/blob/3ec13bd35ebd88286143148b1d637e56e11187ed/deployments/mainnet/DexT1DeploymentLogic.json#L436

Reply:
i think i should add version

Reply:
yeah, we should have a model per version of the DEX to keep logic isolated and understand the differences. are you planning to do that in this PR?

if this doesn't fix the existing version logic, and we need to use raw logs due to decoding complexities, we can likely sign off on that.

Reply:
oh i see, you added version already, but v2 doesn't exist yet?

Reply:
okay, do you want to keep this PR open to wait for that?

Reply:
i shall add that in a later PR